extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/tracing/vertx/VertxTracerFilter.java

```diff
@@ -0,0 +1,136 @@
+package io.quarkus.opentelemetry.tracing.vertx;
+
+import java.util.Iterator;
+
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.spi.CDI;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanBuilder;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import io.quarkus.opentelemetry.QuarkusContextStorage;
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.http.HttpVersion;
+import io.vertx.ext.web.RoutingContext;
+
+public class VertxTracerFilter implements Handler<RoutingContext> {
+    private static final TextMapPropagator TEXT_MAP_PROPAGATOR = GlobalOpenTelemetry.getPropagators().getTextMapPropagator();
+
+    Tracer tracer;
+    boolean initialized = false;
+
+    @Override
+    public void handle(RoutingContext routingContext) {
+        // Needed because CDI Request Scope not active yet
+        QuarkusContextStorage.INSTANCE.setRoutingContext(routingContext);
+
+        if (!initialized) {
+            Instance<Tracer> tracerInstance = CDI.current().select(Tracer.class);
+            if (tracerInstance.isResolvable()) {
+                tracer = tracerInstance.get();
```

Comment:
Will you ever be on this path if the Tracer isn't resolvable? You do handle the null below, but I'm wondering if it is necessary. I would assume this filter wouldn't even be defined unless there is a Tracer about..

Reply:
It's unlikely that `tracerInstance.isResolvable()` would be false if the filter is present.

I'm probably being over cautious about a situation where `Tracer` has been replaced by an application developer and it has an issue.